src/useCameraShake.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react'
+import * as THREE from 'three'
+import { useFrame, Camera } from 'react-three-fiber'
+import { makeNoise2D } from 'open-simplex-noise'
+
+type ShakeConfig = typeof defaultConfig
+export type ShakeConfigPartial = Partial<ShakeConfig>
+
+const defaultConfig = {
+  decay: false,
+  decayRate: 0.65,
+  maxYaw: 0.1,
+  maxPitch: 0.1,
+  maxRoll: 0.1,
+  yawFrequency: 1,
+  pitchFrequency: 1,
+  rollFrequency: 1,
+  yawNoiseSeed: 10,
+  pitchNoiseSeed: 20,
+  rollNoiseSeed: 30,
+}
+
+export interface ShakeController {
```

Comment:
lets trim this to a basic `set` & `get` you can doo `add` and `clear` with `set` & `get`.